### PR TITLE
TGP-1667 added audit events maintain profile

### PR DESCRIPTION
**Pull request by AmarkumarJagtap**

### CODE REVIEW

File: app/factories/AuditEventFactory.scala [hunk 1]
```diff
@@ -54,6 +54,26 @@ case class AuditEventFactory() {
...
 
+  def createMaintainProfileEvent(
+    traderProfile: TraderProfile,
+    updatedTraderProfile: TraderProfile,
+    affinityGroup: AffinityGroup
+  )(implicit hc: HeaderCarrier): DataEvent = {
+    val auditDetails = Map(
+      "eori"                -> traderProfile.actorId,
+      "affinityGroup"       -> affinityGroup.toString
+    ) ++
```
Comment (mauro-fanton) [inline app/factories/AuditEventFactory.scala]: Here you do nor need to use Map concatenation, but just add them as element:

```suggestion
     val auditDetails = Map(
      "eori"                -> traderProfile.actorId,
      "affinityGroup"       -> affinityGroup.toString,
      "previousProfile" -> addUpdatedFieldsOfTraderProfile(traderProfile, updatedTraderProfile),
      "currentProfile" -> addUpdatedFieldsOfTraderProfile(updatedTraderProfile, traderProfile)
    )
```

File: test/services/AuditServiceSpec.scala [hunk 2]
```diff
@@ -861,6 +861,73 @@ class AuditServiceSpec extends SpecBase with BeforeAndAfterEach {
...
+      val fakeAuditEvent = DataEvent("source", "type")
+      when(mockAuditFactory.createMaintainProfileEvent(any(), any(), any())(any())).thenReturn(fakeAuditEvent)
+
+      val result =
+        await(auditService.auditMaintainProfile(traderProfile, updatedTraderProfile, AffinityGroup.Individual))
+
+      result mustBe Done
+
+      withClue("Should have supplied the trader profile and affinity group to the factory to create the event") {
+        verify(mockAuditFactory, times(1))
```
Comment (mauro-fanton) [inline test/services/AuditServiceSpec.scala]: just for future reference, yu do not need to add times(1) as this is the default.

Comment (AmarkumarJagtap) [reply test/services/AuditServiceSpec.scala]: thanks Mauro, that's good spot, I have removed from currently added/updated tests but will raise separate ticket to remove from all the test cases.

File: test/controllers/CyaCategorisationControllerSpec.scala [hunk 3]
```diff
@@ -32,8 +32,6 @@ import play.api.inject.bind
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
 import queries.{LongerCommodityQuery, RecordCategorisationsQuery}
-import uk.gov.hmrc.play.bootstrap.binders.RedirectUrl
```
Comment (besscerule) [inline test/controllers/CyaCategorisationControllerSpec.scala:35]: thanks for getting rid of all the unused imports!!! :)

**Review by besscerule — APPROVED** [review]
looks great :)

**Review by mauro-fanton — APPROVED** [review]

**Pull request merged**